### ISSUE TITLE
Adapt fails on a specialized function

### ISSUE THREAD

**Issue by zsx** [metaeducation/ren-c#546]
The following script fails:
```
REBOL []
v: copy []
append-v: specialize 'append [series: v]

adapted-append-v: adapt 'append-v [
    dump series
    value: to integer! value
]

adapted-append-v "10"
dump v
```

with
```
v: => []
series: => "10"
** Script Error: value has no value
** Where: adapted-append-v do catch either either --anonymous-- do trap either --anonymous--
** Near: to integer! value ??
** File: ../src/core/c-eval.c
** Line: 1487
```
So `adapted-append-v` still expects two parameters [series value], and this can be verified by changing to:

```
adapted-append-v v "10"
```

and it gives:
```
series: => []
v: => [10]
```

**Comment by hostilefork** [comment]
So there's four components.

* The *paramlist*.  This is the unique identity of the function, which has ARRAY_FLAG_PARAMLIST set.  It has the exact right number of parameters, and the [0] element holds a canon value of the function itself.

* The *underlying* function.  This is the form of the frame that it has to have for the innermost code (native, whatever) to be able to run as written.  This is also the pointer that bindings are hooked to.  It may have fewer parameters than the main paramlist.  It should have ARRAY_FLAG_PARAMLIST set, and its [0] element should hold a canon value of the function itself.  It may be the same as the paramlist.

* The *facade*.  This array has the same length as the paramlist of the underlying function (and often is just the paramlist of the underlying function)... but may have different data types or parameter properties.  Currently this is used to do tightening of existing functions, but will someday be used to create versions of a function that accepts narrower datatypes, or create a variant of a function that quotes if it didn't before, etc.  It might *not* have ARRAY_FLAG_PARAMLIST set, and if it doesn't, then its [0] element is the function value of the *underlying* function.  (This is how one can quickly navigate to the underlying function; just look at the VAL_FUNC of the [0] element of the facade)

* The *exemplar*.  This is a context whose varlist provides any specializations.  The concept is that it is okay to go ahead and splice in the values from the exemplar even if the specialization is being done at an inner level with adaptations etc. above it, because as far as those adaptations are concerned the parameters aren't there.  Protections to make sure outer adaptations don't see these parameters aren't in place yet, so frames look bigger than they should to some of these functions.  This will be addressed by the `->phase` field in a FRAME! value, which will be used to subset the keylist.

So, what happened here was that ADAPT reached too deeply to the underlying function, and the underlying function was not specialized.  Hence Make_Function() didn't propagate the exemplar.  *(The idea is that you can go ahead and get your frame set up just by looking at the outermost function, to make calling go faster.)*

Long story short: Make_Function() needed to know the specific adaptee, not the underlying function of the adaptee, when inheriting the exemplar.  It was a bit confusing because the parameter was named "underlying", when the meaning of that parameter was really just "one level underneath".  Terminology makes a big difference, suggestions welcome. 

Good for you to know how all this works.  :-)  Kind of interesting, isn't it?

https://github.com/metaeducation/ren-c/commit/48afaadf37a6c71baf9bc2dff7c5215356ee3e2d

**Comment by zsx** [comment]
Thanks for the explanation. I see that the examplar  in the new function [points to the examplar from the interface](https://github.com/metaeducation/ren-c/blob/master/src/core/c-function.c#L1071), and that's what's missing when passing in an inner most function.
